src/wal/src/local_storage_impl/record_encoding.rs

```diff
@@ -172,38 +172,26 @@ impl RecordEncoding {
 
         // Verify version
         ensure!(
-            version == self.version,
+            version == self.expected_version,
             Version {
-                expected: self.version,
+                expected: self.expected_version,
                 given: version
             }
         );
 
-        // Read CRC
         let crc = buf.try_get_u32().context(Decoding)?;
-
-        // Read length
-        let length = buf.try_get_u32().context(Decoding)?;
-        ensure!(
-            length > 0,
-            LengthMismatch {
-                expected: 1usize,
-                actual: 0usize
-            }
-        );
-
-        // Ensure the buf is long enough
-        ensure!(
-            buf.remaining() >= length as usize,
-            LengthMismatch {
-                expected: length as usize,
-                actual: buf.remaining()
-            }
-        );
+        let table_id = buf.try_get_u64().context(Decoding)?;
+        let sequence_num = buf.try_get_u64().context(Decoding)?;
+        let value_length = buf.try_get_u32().context(Decoding)?;
+        let value = buf[0..value_length as usize].to_vec();
 
         // Verify CRC
-        let data = &buf[0..length as usize];
-        let computed_crc = crc32fast::hash(data);
+        let mut h = Hasher::new();
```

Comment:
Change the calculation logic? Can you comment out the calculation logic?

Reply:
fixed